lib/model/message_list.dart

```diff
@@ -478,6 +502,12 @@ class MessageListView with ChangeNotifier, _MessageSequence {
     }
   }
 
+  void messagesRemoved(List<int> messageIds) {
```

Comment:
nit: let's style this as another event-handling method, so:
* name and signature reflect handling an event — so the same as the method on `MessageStoreImpl`
* ordered in the class next to the other event-handling methods, and in the same order the event types are defined — so just above `handleMessageEvent`

The body will be the same, so the guts of it will still be that `_removeMessagesById` defined on `_MessageSequence`. But that way basically it's this class's responsibility to determine that that's what needs to be done.

That `_removeMessagesById` method is also the part that will get reused, possibly with some adaptation, for other operations like muting a subset of messages or moving some messages out of the narrow.